src/common/column/src/fixedsizebinary/builder.rs

```diff
@@ -0,0 +1,201 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Debug;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+use super::FixedSizeBinaryColumn;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct FixedSizeBinaryColumnBuilder {
+    pub data: Vec<u8>,
+    pub value_length: usize,
+}
+
+impl FixedSizeBinaryColumnBuilder {
+    pub fn with_capacity(data_capacity: usize, value_length: usize) -> Self {
+        FixedSizeBinaryColumnBuilder {
+            data: Vec::with_capacity(data_capacity),
+            value_length,
+        }
+    }
+
+    pub fn from_column(col: FixedSizeBinaryColumn) -> Self {
+        FixedSizeBinaryColumnBuilder {
+            data: col.data.make_mut(),
+            value_length: col.value_length,
+        }
+    }
+
+    pub fn from_data(data: Vec<u8>) -> Self {
+        let value_length = data.len();
+        FixedSizeBinaryColumnBuilder { data, value_length }
+    }
+
+    pub fn repeat(scalar: &[u8], n: usize) -> Self {
+        let len = scalar.len();
+        let data = scalar.repeat(n);
+        FixedSizeBinaryColumnBuilder {
+            data,
+            value_length: len,
+        }
+    }
+
+    pub fn repeat_default(n: usize, value_length: usize) -> Self {
+        FixedSizeBinaryColumnBuilder {
+            data: vec![0; n * value_length],
+            value_length,
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        if self.data.is_empty() {
+            0
+        } else {
+            self.data.len() / self.value_length
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.len() <= 1
+    }
+
+    pub fn memory_size(&self) -> usize {
+        self.data.len()
+    }
+
+    pub fn put_u8(&mut self, item: u8) {
+        self.data.push(item);
+    }
+
+    pub fn push_default(&mut self) {
+        self.data.extend_from_slice(&vec![0; self.value_length]);
+    }
+
+    pub fn put_char(&mut self, item: char) {
+        self.data
+            .extend_from_slice(item.encode_utf8(&mut [0; 4]).as_bytes());
+    }
+
+    #[inline]
+    pub fn put_str(&mut self, item: &str) {
+        debug_assert!(self.value_length == item.len());
+        self.data.extend_from_slice(item.as_bytes());
+    }
+
+    #[inline]
+    pub fn put_slice(&mut self, item: &[u8]) {
+        debug_assert!(self.value_length == item.len());
+        self.data.extend_from_slice(item);
+    }
+
+    #[inline]
+    pub fn commit_row(&mut self) {
```

Comment:
seems don't need a `commit_row` function.

Reply:
I don't figure out the reason.

Reply:
drop it.